src/lib/src/component/navigation/navigation.component.ts

```diff
@@ -71,32 +74,49 @@ export class NavigationComponent implements OnInit {
       this.retrieveWorkspaceRoot();
       this.changeDetectorRef.detectChanges();
     });
-    this.messagingService.subscribe(events.NAVIGATION_REFRESH, element => {
-      this.retrieveWorkspaceRoot();
+    this.messagingService.subscribe(events.NAVIGATION_CREATED, payload => {
+      this.handleNavigationCreated(payload);
     });
     this.messagingService.subscribe(events.NAVIGATION_SELECT, element => {
       this.uiState.selectedElement = element as WorkspaceElement;
       this.changeDetectorRef.detectChanges();
     });
   }
 
+  handleNavigationCreated(payload: any): void {
+    this.retrieveWorkspaceRoot().then(() => {
+      this.revealElement(payload.path);
+      this.selectElement(payload.path);
+    });
+  }
+
   newElement(type: string): void {
     let selectedElement = this.uiState.selectedElement;
     if (selectedElement) {
-      if (selectedElement.type == ElementType.Folder) {
+      if (selectedElement.type === ElementType.Folder) {
         this.uiState.setExpanded(selectedElement.path, true);
       }
     }
     this.uiState.newElementRequest = {
       selectedElement: selectedElement,
       type: type
-    }
+    };
     this.changeDetectorRef.detectChanges();
   }
 
   collapseAll(): void {
     this.uiState.clearExpanded();
-    this.uiState.setExpanded(this.workspaceRoot.path, true);
+    this.uiState.setExpanded(this.workspace.root.path, true);
+  }
+
+  revealElement(path: string): void {
+    let subpaths = this.workspace.getSubpaths(path);
+    subpaths.forEach(subpath => this.uiState.setExpanded(subpath, true));
```

Comment:
does setting the expanded flag of the last subpath display all the sub element, too?
asked differently: shouldn't the expanded flag be set except on the element itself (last subpath)?

Reply:
Hm maybe the naming is misleading here... subpath means "part of the path" - not sub element.
So `a/b/c` consists of the subpaths `a/`, `a/b` and `a/b/c`. Don't have a better name yet... path fragment is not good either as I would expect a fragment to be only a part like `b/`